pulp_smash/tests/pulp3/file/api_v3/test_sync.py

```diff
@@ -85,3 +86,43 @@ def do_test(self, download_policy, sync_mode):
         sync_repo(self.cfg, importer)
         repo = client.get(repo['_href'])
         self.assertNotEqual(latest_version_href, repo['_latest_version_href'])
+
+
+class SyncChangeRepoVersionTestCase(unittest.TestCase):
+    """Verify whether sync of repository updates repository version."""
+
+    def test_all(self):
+        """Verify whether the sync of a repository updates its version.
+
+        This test explores the design choice stated in the `Pulp #3308`_ that a
+        new repository version is created even if the sync does not add or
+        remove any content units. Even without any changes to the importer if a
+        new sync occurs, a new repository version is created.
+
+        .. _Pulp #3308: https://pulp.plan.io/issues/3308
+
+        Do the following:
+
+        1. Create a repository, and an importer.
+        2. Sync the repository an arbitrary number of times.
+        3. Verify that the repository version is equal to the previous number
+           of syncs.
+        """
+        cfg = config.get_config()
+        client = api.Client(cfg, api.json_handler)
+        client.request_kwargs['auth'] = get_auth()
+        repo = client.post(REPO_PATH, gen_repo())
+        self.addCleanup(client.delete, repo['_href'])
+        body = gen_importer(repo)
```

Comment:
This function needs to return importers that have valid download policies before this test can be merged.

Reply:
Logic related to download policy was added to [get_importer_down_policy](https://github.com/PulpQE/pulp-smash/blob/14c9dc7a1a0edb21bf91904d92b9431c06505e70/pulp_smash/tests/pulp3/file/api_v3/utils.py#L12).